### PR TITLE
chore: Update actions in release workflow

### LINKED ISSUE #275: `release.yml` uses deprecated `Node.js 16 actions`
[`release.yml` uses deprecated `Node.js 16 actions`](https://github.com/amannn/action-semantic-pull-request/actions/runs/9710385301).

### PR DESCRIPTION
**Pull request by hamirmahal**
Fixes #275

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ jobs:
   main:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
```
Comment (hamirmahal) [inline .github/workflows/release.yml:18]: `v4` is the latest major version of both of these actions, at the time of this writing.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: 'Release'
+name: "Release"
```
Comment (amannn) [inline .github/workflows/release.yml:1]:
```suggestion
name: 'Release'
```

**Review by amannn — APPROVED** [review]
Awesome, thanks a lot!

**Comment by hamirmahal** [comment]
You're welcome!

**Pull request merged**